### ISSUE TITLE
cargo doc-upload fails but build succeeds?

### ISSUE THREAD

**Issue by gnzlbg** [roblabla/cargo-travis#29]
See https://travis-ci.org/gnzlbg/slice_deque/jobs/330604987#L1460

I get this error:

```shell
GitHub Personal Access Token was not provided in $GH_TOKEN or --token
Falling back to using the SSH endpoint
Cloning into 'target/doc-upload'...
Warning: Permanently added the RSA host key for IP address '192.30.253.113' to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.
```

yet my build succeeds? :/

Also, I have a GH_TOKEN in my travis.yml file, any idea what could be going wrong?

**Comment by CAD97** [comment]
Build passes because the command is returning an exit code of 0, even though the Git communication fails. This is a bug.

Given the `env: global: - secure:`, I can't actually see your env (this is a good thing).

But according to your logs, GH_PAGES isn't set: https://travis-ci.org/gnzlbg/slice_deque/jobs/330604987#L489-L490

```
Setting environment variables from .travis.yml
$ export WZI=[secure]
```

**Comment by CAD97** [comment]
Similarly, `cargo doc-upload` exits with `1` if there is nothing to commit.

**Comment by gnzlbg** [comment]
> But according to your logs, GH_TOKEN isn't set

Thanks, I've fixed this. I was trying to use the GH_TOKEN from another project without using `travis encrypt` for this one -.-

**Comment by gnzlbg** [comment]
> Build passes because the command is returning an exit code of 0, even though the Git communication fails. This is a bug. 

Should I open a different issue for this?

**Comment by gnzlbg** [comment]
Now I am getting this error after managing to get docs working for a while: 

https://travis-ci.org/gnzlbg/slice_deque/jobs/335382083#L1513

**Comment by roblabla** [comment]
The original problem of this issue (doc-upload succeeds despite auth fails) is fixed with #63.
